waku/v2/waku_archive/driver/sqlite_driver/sqlite_driver.nim

```diff
@@ -83,30 +73,25 @@ method put*(s: SqliteDriver, pubsubTopic: PubsubTopic, message: WakuMessage, dig
     int64(message.version),        # version
     message.timestamp              # senderTimestamp
   ))
-  if res.isErr():
-    return err("message insert failed: " & res.error)
-
-  ok()
 
-
-method getAllMessages*(s: SqliteDriver):  ArchiveDriverResult[seq[ArchiveRow]] =
+method getAllMessages*(s: SqliteDriver):
+                       Future[ArchiveDriverResult[seq[ArchiveRow]]] {.async.} =
   ## Retrieve all messages from the store.
-  s.db.selectAllMessages()
-
-
-method getMessages*(
-  s: SqliteDriver,
-  contentTopic: seq[ContentTopic] = @[],
-  pubsubTopic = none(PubsubTopic),
-  cursor = none(ArchiveCursor),
-  startTime = none(Timestamp),
-  endTime = none(Timestamp),
-  maxPageSize = DefaultPageSize,
-  ascendingOrder = true
-): ArchiveDriverResult[seq[ArchiveRow]] =
+  return s.db.selectAllMessages()
+
+method getMessages*(s: SqliteDriver,
+                    contentTopic: seq[ContentTopic] = @[],
+                    pubsubTopic = none(PubsubTopic),
+                    cursor = none(ArchiveCursor),
+                    startTime = none(Timestamp),
+                    endTime = none(Timestamp),
+                    maxPageSize = DefaultPageSize,
+                    ascendingOrder = true):
+                    Future[ArchiveDriverResult[seq[ArchiveRow]]] {.async.} =
+
   let cursor = cursor.map(toDbCursor)
 
-  let rows = ?s.db.selectMessagesByHistoryQueryWithLimit(
+  return s.db.selectMessagesByHistoryQueryWithLimit(
```

Comment:
As above, I'd keep the indirection here with assigning this to a variable first to have the SQLite-specific failure and return cleanly separated within the SqliteDriver module.